src/imputation/tmi_imputation.py

```diff
@@ -405,6 +405,26 @@ def apply_tmi(df, target_variables, mean_dict):
     return final_df
 
 
+def calculate_totals(df):
```

Comment:
What is needed here is to also keep the original values where imputation does not take place. 
You might like to look at how this is done in the the rest of tmi_imputation.py

Reply:
Ignore this, sorry, it's already done